packages/Search/src/DTK_FineSearch_decl.hpp

```diff
@@ -0,0 +1,80 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_FINE_SEARCH_DECL_HPP
+#define DTK_FINE_SEARCH_DECL_HPP
+
+#include "DTK_ConfigDefs.hpp"
+#include <Kokkos_Core.hpp>
+#include <Shards_CellTopology.hpp>
+
+namespace DataTransferKit
+{
+template <typename DeviceType>
+class FineSearch
+{
+  public:
+    /**
+     * Performs the local search .
+     *    @param[out] reference_points The coordinates of the points in the
```

Comment:
Add dimensionalities descriptions, like (C,D,N) and friends. I thought I saw that in the original patch?